KubeArmor/core/unorchestratedUpdates.go

```diff
@@ -503,6 +523,12 @@ func (dm *KubeArmorDaemon) ParseAndUpdateContainerSecurityPolicy(event tp.K8sKub
 			// add the endpoint into the endpoint list
 			dm.EndPoints = append(dm.EndPoints, newPoint)
 		} else {
+			for idx, endPoint := range dm.EndPoints {
+				if kl.MatchIdentities(secPolicy.Spec.Selector.Identities, endPoint.Identities) {
```

Comment:
Is there a reason to do this step again, as we have already updated value of `i` above at [line](https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R431).

Reply:
This is because if the dm.Endpoints object is deleted by the following process, the value of i, which indicates the index of the object, has changed.

```
dm.EndPoints = append(dm.EndPoints[:idx], dm.EndPoints[idx+1:]...)
```

https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R443


If i is not recalculated, an out-of-array reference error will occur in the following operations.

```
dm.endpoints [i] = NewPoint
```

https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R532

Reply:
🤔 That makes sense. So can we assign the value of i after [line 458](https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R458)
and remove the loop from [here 431](https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R431) 
this would also remove the need to reassign the value to `i` from [here 526 ]( https://github.com/kubearmor/KubeArmor/pull/1406/files#diff-168b945c99cdc3669f72461c0f4913020872a0e7ff4222676b356125676a4c64R526).